src/main/scala/com/cloudera/sparkts/TimeSeriesUtils.scala

```diff
@@ -198,6 +195,36 @@ private[sparkts] object TimeSeriesUtils {
     }
   }
 
+  def rebaserGeneric(
```

Comment:
Can you add a comment describing why we don't use this in all situations?  I.e. that it can be O(n log n) at times when we can achieve O(n).  And that it would be possible to include more efficient implementations that achieve O(n).